frontend/test/test_oqd/oqd/test_oqd_database_managers.py

```diff
@@ -0,0 +1,536 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for parsing and loading the configuration files for an Open Quantum Design (OQD)
+trapped-ion quantum computer device.
+"""
+
+# pylint: disable=import-outside-toplevel
+
+import math
+import pathlib
+import textwrap
+
+import pytest
+
+FRONTEND_ROOT_PATH = pathlib.Path(__file__).parent.parent.parent.parent
+OQD_SRC_DIR = FRONTEND_ROOT_PATH / "catalyst/third_party/oqd/src/"
+OQD_TEST_DIR = FRONTEND_ROOT_PATH / "test/test_oqd/oqd/"
+
+
+class TestOQDDeviceDatabase:
+    """Test suite for the OQDDeviceDatabase class."""
+
+    def test_from_toml_production(self):
+        """
+        Tests that the OQDDeviceDatabase.from_toml method can load the device properties from the
+        oqd_device_parameters.toml file used in production.
+        """
+        from catalyst.third_party.oqd import OQDDeviceDatabase
+
+        device_database = OQDDeviceDatabase.from_toml(OQD_SRC_DIR / "oqd_device_parameters.toml")
+        assert device_database.parameters
+
+    def test_from_toml_string(self):
+        """
+        Tests that the OQDDeviceDatabase.from_toml method can load the device properties from a
+        TOML document string.
+        """
+        from catalyst.third_party.oqd import OQDDeviceDatabase
```

Comment:
Is there a reason why we are importing the databases per test, instead of once at the top of the file?

Reply:
I was following the pattern used in the OQC device tests, e.g. here https://github.com/PennyLaneAI/catalyst/blob/main/frontend/test/test_oqc/oqc/test_oqc_device.py#L34-L38.

I'm not sure what the reason for this was, maybe to explicitly test that the classes can be imported? I think it's fine to move the imports to the top of the file here, in fact it might even save a bit of runtime only importing each class once.

Reply:
I mean, import failures will still be reported if they are at the top of the file right? But if this is an existing pattern I'm happy with leaving it as is. 

Maybe cc @dime10 @mlxd as to some best practice questions here.

Reply:
Chatted with @mlxd , let's try importing this at top level.

Reply:
Done: 5576c728